### PR TITLE
[12.0][MIG] web_environment_ribbon

### PR DESCRIPTION
**Pull request by tarteo**
I've updated the readme and fixed some jslint issues.

### CODE REVIEW

**Review by lmignon — APPROVED** [review]
@tarteo Thank you for the migration. 
A good practice is to put the changes related to code formatting in a separate commit to ease the review and don't mix changes for the migrations with cosmetic stuff. Moreover, code formatting in JS is a matter of personal preference and AFAIK no precise guidelines exists into OCA.  Even if I'm sometimes tempted to change the code formatting into some addons (because my IDE do it for me), I avoid this kind of change.

**Comment by tarteo** [comment]
@lmignon I only fixed things that give warnings in travis. Next time I'll do it in a seperate commit. Thanks!

**Review by simahawk — APPROVED** [review]
LG

**Review by ThomasBinsfeld — APPROVED** [review]

File: web_environment_ribbon/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+* You can change the ribbon's name ("TEST") by editing the default system
+  parameter "ribbon.name" (in the menu Settings > Parameters > System
+  Parameters) To hide the ribbon, set this parameter to "False" or delete it.
```
Comment (SimoRubi) [inline web_environment_ribbon/readme/CONFIGURE.rst:3]: Setting all the ribbon parameters to False does not hide the ribbon:
image

Deleting them results in 
```
Odoo Server Error
Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 653, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 312, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 695, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 344, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 337, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 938, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 517, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/OCB-12.0/addons/web/controllers/main.py", line 955, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/odoo/OCB-12.0/addons/web/controllers/main.py", line 947, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 745, in call_kw
    return _call_kw_model(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 718, in _call_kw_model
    result = method(recs, *args, **kwargs)
  File "/home/odoo/build/OCA/web/web_environment_ribbon/models/web_environment_ribbon_backend.py", line 32, in get_environment_ribbon
    name = self._prepare_ribbon_name()
  File "/home/odoo/build/OCA/web/web_environment_ribbon/models/web_environment_ribbon_backend.py", line 23, in _prepare_ribbon_name
    return name_tmpl.format(**vals)
AttributeError: 'bool' object has no attribute 'format'
```

**Review by SimoRubi — CHANGES_REQUESTED** [review]
Thanks for your contribution!
Tested on runbot

File: web_environment_ribbon/__init__.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Copyright 2015 Francesco OpenCode Apruzzese <cescoap@gmail.com>
```
Comment (nikul-serpentcs) [inline web_environment_ribbon/__init__.py]: Remove Copyright from all `__init__` file

**Review by nikul-serpentcs — COMMENTED** [review]
Small Change

File: web_environment_ribbon/readme/CONFIGURE.rst (hunk 1 above)
Comment (tarteo) [reply web_environment_ribbon/readme/CONFIGURE.rst:3]: Thanks for your review! I didn't noticed it, but I changed the behavior to hide the ribbon if the system parameter does not exists. Can you test it again, please and reconsider your review. :)

File: web_environment_ribbon/__init__.py (hunk 2 above)
Comment (tarteo) [reply web_environment_ribbon/__init__.py]: I removed the copyright from the __init__.py files. Can you reconsider your review, please?

**Comment by tarteo** [comment]
The red cross seems to be related to failing travis. We'll have to wait and rebuild.

**Comment by tafaRU** [comment]
> The red cross seems to be related to failing travis.

It's due to odoo branch `12.0` which isn't been created yet!

File: web_environment_ribbon/static/src/js/ribbon.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        $('body').append(ribbon);
+        ribbon.hide();
+        // Get ribbon data from backend
+        rpc.query({
+            model: 'web.environment.ribbon.backend',
+            method: 'get_environment_ribbon',
+        }).then(
+            function (ribbon_data) {
+                // Ribbon name
+                console.log(ribbon_data);
```
Comment (SimoRubi) [inline web_environment_ribbon/static/src/js/ribbon.js]: @tarteo Are you logging in the console on purpose or is this just some of your debugging code?

**Review by SimoRubi — APPROVED** [review]
Tested on runbot, just check https://github.com/OCA/web/pull/1058#pullrequestreview-161131789

File: web_environment_ribbon/static/src/js/ribbon.js (hunk 3 above)
Comment (tarteo) [reply web_environment_ribbon/static/src/js/ribbon.js]: It was for debugging so I removed it. Thanks!

File: web_environment_ribbon/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    'category': 'Web',
+    'author': 'Francesco OpenCode Apruzzese, '
+              'Tecnativa, '
+              'Odoo Community Association (OCA)',
+    'website': 'https://github.com/OCA/web',
+    'license': 'AGPL-3',
+    'depends': [
+        'web',
+        ],
+    'data': [
```
Comment (nikul-serpentcs) [inline web_environment_ribbon/__manifest__.py]: @tarteo IMO `data` Sequence as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L29)

**Review by nikul-serpentcs — COMMENTED** [review]
Small Change

File: web_environment_ribbon/__manifest__.py [hunk 5]
```diff
@@ -6,22 +6,20 @@
...
         ],
-    "data": [
+    'data': [
         'view/base_view.xml',
         'data/ribbon_data.xml',
         ],
-    "update_xml": [],
-    "demo_xml": [],
-    "auto_install": False,
+    'auto_install': False,
```
Comment (yajo) [inline web_environment_ribbon/__manifest__.py]: Please leave quotes as they were.

File: web_environment_ribbon/__manifest__.py [hunk 6]
```diff
@@ -6,22 +6,20 @@
...
               'Tecnativa, '
               'Odoo Community Association (OCA)',
     'website': 'https://github.com/OCA/web',
     'license': 'AGPL-3',
-    "depends": [
+    'depends': [
         'web',
         ],
-    "data": [
+    'data': [
```
Comment (yajo) [inline web_environment_ribbon/__manifest__.py]: Please leave quotes as they were.

File: web_environment_ribbon/__manifest__.py [hunk 7]
```diff
@@ -6,22 +6,20 @@
...
-    'version': '11.0.1.0.2',
+    'version': '12.0.1.0.0',
     'category': 'Web',
     'author': 'Francesco OpenCode Apruzzese, '
               'Tecnativa, '
               'Odoo Community Association (OCA)',
     'website': 'https://github.com/OCA/web',
     'license': 'AGPL-3',
-    "depends": [
+    'depends': [
```
Comment (yajo) [inline web_environment_ribbon/__manifest__.py]: Please leave quotes as they were.

**Review by yajo — APPROVED** [review]

**Comment by yajo** [comment]
Please, once you do the fix above :point_up: squash all migration-related commits (not the old history) in one or two commits, to merge. Thanks.

File: web_environment_ribbon/__manifest__.py (hunk 4 above)
Comment (tarteo) [reply web_environment_ribbon/__manifest__.py]: Sure

**Comment by tarteo** [comment]
Thank you for all the reviews!

**Pull request merged**